packages/in_app_purchase/in_app_purchase_ios/lib/src/store_kit_wrappers/sk_product_wrapper.dart

```diff
@@ -241,7 +241,8 @@ class SKProductWrapper {
     required this.price,
     this.subscriptionPeriod,
     this.introductoryPrice,
-  });
+    List<SKProductDiscountWrapper>? discounts,
```

Comment:
```suggestion
    this.discounts = <SKProductDiscountWrapper>[],
```
And remove the next line.

Reply:
I remember there's a way to indicate default value to be empty array in jsonSerializable.

Reply:
Updating `List<SKProductDiscountWrapper>? discounts,` to `this.discounts = <SKProductDiscountWrapper>[],` will not work as the default value is not a constant. The following error is generated:

```
The default value of an optional parameter must be constant. dart(non_constant_default_value)
```

That is why I resolved it like this. The field is already configured to use an empty array as default value in jsonSerializable, however this is done on field level not in the constructor:

```dart
  @JsonKey(defaultValue: <SKProductDiscountWrapper>[])
  final List<SKProductDiscountWrapper> discounts;
```

I am not aware you can add additional attributes on constructor parameters.

Reply:
now you've updated the filed to be none-optional, will `this.discounts = const <SKProductDiscountWrapper>[],` not work? The parameter doesn't have to be optional, right?